### DISCUSSION TITLE
NEXTAUTH_URL not taken into account

### DISCUSSION THREAD

**Discussion by leonard-henriquez** [nextauthjs/next-auth#5844] [Help]
### Environment

```
  System:
    OS: macOS 12.3
    CPU: (10) arm64 Apple M1 Pro
    Memory: 73.66 MB / 32.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 18.12.0 - ~/.asdf/installs/nodejs/lts/bin/node
    npm: 8.19.2 - ~/.asdf/plugins/nodejs/shims/npm
  Browsers:
    Chrome: 107.0.5304.87
    Firefox Developer Edition: 107.0
    Safari: 15.4
  npmPackages:
    next: ^13.0.0 => 13.0.0
    next-auth: ^4.15.0 => 4.15.0
    react: ^18.2.0 => 18.2.0
```

### Reproduction URL

only depend on env variable

## Context

I am trying to use Next Auth with an external API made with Nest (express.js).
I found some information that helped me to acheive this goal here:
- https://github.com/nextauthjs/next-auth/issues/2294
- https://github.com/nextauthjs/next-auth/issues/531

Now, I need `next-auth` to make API calls to this external API (`localhost:3001`).

## Problem

I tried to add `NEXTAUTH_URL='http://localhost:3001'` or even `NEXTAUTH_URL_INTERNAL`  in my .env.
Those environment variables seem to be accessible by my Next application and yet, the API call are made to `localhost:3000`. However redirections are made to url starting with `localhost:3001`.

Is it possible to do the exact opposite? (e.g. send API request to `localhost:3001` and keep redirections on `localhost:3000`)

### How to reproduce

1. add `NEXTAUTH_URL='http://localhost:3001'` to your `.env`
2. launch server
3. go to login page
4. check network requests in the chrome debugger and observe that API call made by next-auth are made to `http://localhost:3000` (not 300**1**)

### Expected behavior

- I would expect NEXTAUTH_URL to change the url of the API calls
- I would expect clearer explanation on `NEXTAUTH_URL` and `NEXTAUTH_URL_INTERNAL` in the documentation

**Comment by ThangHuuVu** [comment]
Would you be able to provide a minimal reproduction here? 🙏

**Comment by leonard-henriquez** [comment]
Sorry, unfortunately I stopped using NextAuth for this reason...
